source/COMRegistrationFixes/__init__.py

```diff
@@ -1,81 +1,126 @@
+SYSTEM_ROOT = os.path.expandvars("%SYSTEMROOT%")
+SYSTEM32 = os.path.join(SYSTEM_ROOT, "System32")
+SYSNATIVE = os.path.join(SYSTEM_ROOT, "Sysnative")  # Virtual folder for reaching 64-bit exes from 32-bit apps
+SYSTEM_DRIVE = os.path.expandvars("%SYSTEMDRIVE%\\")
+PROGRAM_FILES = os.path.join(SYSTEM_DRIVE, "Program Files")
+PROGRAM_FILES_X86 = os.path.join(SYSTEM_DRIVE, "Program Files (x86)")
+
 
-def registerServer(fileName,wow64=False):
+def register32bitServer(fileName: str) -> None:
+	"""Registers the COM proxy dll with the given file name, using the 32-bit version of regsvr32.
+	Note: this function is valid while NVDA remains a 32-bit app. Re-evaluate if we move to 64-bit.
+	@param fileName: the path to the dll
+	@type fileName: str
 	"""
-	Registers the COM proxy dll with the given file name
-	Using regsvr32.
+	# Points to the 32-bit version, on Windows 32-bit or 64-bit.
+	regsvr32 = os.path.join(SYSTEM32, "regsvr32.exe")
+	try:
+		subprocess.check_call([regsvr32, "/s", fileName])
+	except subprocess.CalledProcessError as e:
+		log.error(f"Error registering {fileName} in a 32-bit context: {e}")
+	else:
+		log.debug(f"Registered {fileName} in a 32-bit context.")
+
+
+def register64bitServer(fileName: str) -> None:
+	"""Registers the COM proxy dll with the given file name, using the 64-bit version of regsvr64.
+	Note: this function is valid while NVDA remains a 32-bit app. Re-evaluate if we move to 64-bit.
 	@param fileName: the path to the dll
 	@type fileName: str
-	@param wow64: If true then the 32 bit (wow64) version of regsvr32 will be used.
-	@type wow64: bool
 	"""
-	regsvr32=os.path.join(sysWow64 if wow64 else system32,'regsvr32.exe')
+	# SysWOW64 provides a virtual directory to allow 32-bit programs to reach 64-bit executables.
+	regsvr32 = os.path.join(SYSNATIVE, "regsvr32.exe")
 	try:
-		subprocess.check_call([regsvr32,'/s',fileName])
+		subprocess.check_call([regsvr32, "/s", fileName])
 	except subprocess.CalledProcessError as e:
-		log.error("Error registering %s, %s"%(fileName,e))
+		log.error(f"Error registering {fileName} in a 64-bit context: {e}")
 	else:
-		log.debug("Registered %s"%fileName)
+		log.debug(f"Registered {fileName} in a 64-bit context.")
+
 
-def applyRegistryPatch(fileName,wow64=False):
+def apply32bitRegistryPatch(fileName: str) -> None:
+	"""Applies the registry patch with the given file name, using 32-bit regExe.
+	Note: this function is valid while NVDA remains a 32-bit app. Re-evaluate if we move to 64-bit.
+	@param fileName: the path to the .reg file
+	@type fileName: str
 	"""
-	Applies the registry patch with the given file name
-	using regedit.
+	if not os.path.isfile(fileName):
+		raise FileNotFoundError(f"Cannot apply 32-bit registry patch: {fileName} not found.")
+	# On 32-bit systems, reg.exe is in System32. On 64-bit systems, SysWOW64 will redirect to 32-bit version.
+	regExe = os.path.join(SYSTEM_ROOT, "System32", "reg.exe")
+	try:
+		subprocess.check_call([regExe, "import", fileName])

```

Comment:
For each run of `reg.exe` a console window is shown briefly it seems. Have you considered calling these with a `STARTUPINFO` structure and `wShowWindow` set to `SW_HIDE`?

Reply:
I'll be running some tests with that option later today.
Thanks.

Reply:
@lukaszgo1 I have done this. I also did it for the regsvr32.exe calls.